server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/operationhandler/EcoreLabelEditOperationHandler.java

```diff
@@ -95,16 +97,18 @@ public void executeOperation(ApplyLabelEditOperation editLabelOperation, GModelS
 					Optional<EClassifier> datatype = parseStringToEType(split[1].trim(),
 							EcoreModelState.getResourceManager(graphicalModelState));
 					if (datatype.isPresent()) {
-						if (!modelAccess.setAttributeType(EcoreModelState.getModelState(graphicalModelState), attribute_semantic, datatype.get())) {
+						if (!modelAccess.setAttributeType(EcoreModelState.getModelState(graphicalModelState),
```

Comment:
Separating the setAttributeType command from the setAttributeName means that we have to undo 2 changes when editing both the name and the type at the same time, which is a bit confusing.

I think it's fine to keep these Model Access methods separate, but we should consider a mechanism to call these methods with a transaction mechanism, and ensure the resulting commands are regrouped into the same compound (Without having a "SetNameAndType" method, because otherwise we'll need to create many methods for the different possible combinations).

Reply:
I updated the modelaccess method and now it's possible to set the attribute name/type as a compound command.